src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTask.cs

```diff
@@ -462,7 +462,6 @@ private List<string> GetFilesForRestore(string backupLocation)
             var options = new DatabaseSmugglerOptionsServerSide
             {
                 AuthorizationStatus = AuthorizationStatus.DatabaseAdmin,
-                OperateOnTypes = ~(DatabaseItemType.CompareExchange | DatabaseItemType.Identities),
                 SkipRevisionCreation = true
             };
 
```

Comment:
1) In case of backup failure, make sure that all the compare exchange & identities are deleted (in case it happened before we saved the database record)
2) RestoreFromLastFile isn't needed anymore since you already imported everything from all the files.

Reply:
@grisha-kotler  how come we can create a compare-exchange values before there is a database record?

Reply:
I think we should do it the other way around. 
But once that happen, I don't think we should delete partial failure, let the user handle this

Reply:
We can't access the compare exchange and the identities in a partial failure because the database doesn't exist. @maximburyak we added a new property IsFromBackup to bypass that.

We thought about doing something better:
1) Add a property in the DatabaseRecord: State: Normal/RestoreInProgress.
2) Save the database record, Disabled and with RestoreInProgress before starting the backup (and prevent access to it while restoring)
3) If we fail we send DeleteDatabaseCommand (that will delete everything that is needed)
4) After a successful restore, enable the database.

Reply:
Just realized that we can't restore/import compare exchange or identities when the cluster is down.
We even can't restore the database record itself.

I like the flow @grisha-kotler described, but maybe without the additional property and rely only on the Disabled.
 
As a side note: In the studio (not for this issue) we should gray-out the import options of compare-exchange and identities when there is not cluster.

Reply:
The restore process should be:


* Create the database record on the local machine, marked disabled
* Restore cmp xchg, identities, etc
* Restore the database itself, as usual
 
Error midway, abort as is and let the admin clean it up

Reply:
@karmeli87 replying only on the disabled will allow the user to enable the database

Reply:
So what? If user want to shoot own foot, let them